ElrondNetwork/elrond-go
# Open more dbs if necessary

## Description of the reasoning behind the pull request (what feature was missing / how the problem was manifesting itself / what was the motive behind the refactoring)
- In order to have the complete state, we need 2 trie dbs marked as active. When restarting the node, we have to be sure that a minimum of 2 trie dbs marked as active are also opened, else the node will not have access to the entire state. 
  
## Proposed Changes
- When a node is restarting, opend trie dbs until there are a minimum of 2 that are marked as active (if the trie was synced, only one db marked as active is needed)

## Testing procedure
- Restart a node after there is a snapshot that takes more than one epoch, and check that the dbs are opened and closed correctly (this will be tested on a different branch, in which a time.Sleep() is added in order to force the snapshot process to take more than an epoch)
